Feature/flume light

philip-morlier opened the pull request.

philip-morlier committed:
Initial commit of flume light, early sketches.

philip-morlier committed:
http client set outside of function, heaveyserver set in config.

philip-morlier committed:
Several changes to the project to facilitate flume light.

A heavy package has been added to enable http requests to a provided "flume-heavy" instance.

The api has been updated such that some of the methods can be routed to heavy if necessary. (This needs more work to be complete)

The consumer now has a process by which to check if an istance is heavy or light and direct the block consumption accordingly.

The config has been elaborated to include a heavy server string.

philip-morlier committed:
Remving commented code.

philip-morlier committed:
light / heavy conditions added to transactions api.

philip-morlier committed:
Merge branch 'develop' into feature/flume-light

philip-morlier committed:
merged origin develop, further api development.

philip-morlier committed:
Current flume light api work.

philip-morlier committed:
Flume light api configuration complete.

Flume light consumption configuration complete.

The api will need to be revisited to accomodate for accurate metrics gathering.

philip-morlier committed:
Further api implimentations:
block checks added to MaxPriorityFeePerGas as well as GasPrice. Metrics gathering completed.

Heavy server custom error for testing implimented.

AusIV commented on api/block.go:
```diff
@@ -103,7 +156,28 @@ func (api *BlockAPI) GetBlockByHash(ctx context.Context, blockHash types.Hash, i
+		log.Debug("eth_getBlockTransactionCountByNumber sent to flume heavy")
+		missMeter.Mark(1)
+		gtcbnMissMeter.Mark(1)
+		count, err := heavy.CallHeavy[hexutil.Uint64](ctx, api.cfg.HeavyServer, "eth_getBlockTransactionCountByNumber", hexutil.Uint64(blockNumber))
```
I think you want to leave blockNumber as a vm.BlockNumber here, which could leave in values like "latest", which is not going to convert to a hexutil.Uint64 cleanly.

AusIV commented on plugins/packages/polygon/migrator_indexer.go:
```diff
@@ -56,6 +56,8 @@ func (pg *PolygonIndexer) Index(pb *delivery.PendingBatch) ([]string, error) {
 		)) 
 	}
 
+	log.Info("sb", "sb", statements)
```
Can we either take this out or make it debug level?

philip-morlier committed:
Merge remote-tracking branch 'origin/develop' into feature/flume-light

Commit:
Put the cardinal_offsets table in the blocks database

Current code was putting the cardinal_offsets table in the in-memory
database, which meant it wasn't around for resumption.

philip-morlier committed:
Merge pull request #31 from openrelayxyz/bugfix/cardinal-offsets

Put the cardinal_offsets table in the blocks database

philip-morlier committed:
Merge branch 'feature/flume-light' of github.com:openrelayxyz/cardinal-flume into feature/flume-light

philip-morlier committed:
Consumer modified to allow for stand alone flume light instances, flume light w heavy counterpart instances, and heavy instances.

Removed un-needed print statement from ploygon indexer.

Removed casting from blocks passed to heavy server per pr comment.

philip-morlier committed:
CallHeavy test created, completed, and functional.

txDataPresent augmented to check mempool.

GetTransactionReceiptsByBlockNumber altered to accept vm.BlockNumber.

philip-morlier committed:
formatted all files in root
